torch/distributed/fsdp/sharded_grad_scaler.py

```diff
@@ -288,28 +302,25 @@ def unscale_(self, optimizer: torch.optim.Optimizer) -> None:
         if future_handles:
             torch.futures.wait_all(future_handles)
 
-    def step(
-        self, optimizer: torch.optim.Optimizer, *args, **kwargs
-    ) -> Optional[float]:
-        return super().step(optimizer, *args, **kwargs)
-
-    def _amp_update_scale_cpu_(self, found_inf) -> None:
+    def _amp_update_scale_cpu_(self, found_inf: torch.Tensor) -> None:
         """
         If found_inf is 1.0 (True), then scale is multiplied by backoff_factor and growth_tracker is set to zero.
         Otherwise, scale is multiplied by the growth factor when the growth interval is reached.
         """
+        assert self._scale is not None and self._growth_tracker is not None
+
         if found_inf.item() >= 1.0:
-            self._scale *= self._backoff_factor  # type: ignore[arg-type]
-            self._growth_tracker = 0
+            self._scale *= self._backoff_factor
+            self._growth_tracker.fill_(0)
         else:
-            successful = self._growth_tracker + 1  # type: ignore[operator]
-            if successful == self._growth_interval:  # type: ignore[arg-type]
-                self._scale *= self._growth_factor  # type: ignore[arg-type]
-                self._growth_tracker = 0
+            successful = self._growth_tracker + 1
+            if successful == self._growth_interval:
+                self._scale *= self._growth_factor
+                self._growth_tracker.fill_(0)
             else:
                 self._growth_tracker = successful
 
-    def update(self, new_scale: Optional[Union[float, FloatTensor]] = None) -> None:
+    def update(self, new_scale: Optional[Union[float, torch.Tensor]] = None) -> None:
```

Comment:
`torch.cuda.FloatTensor` is not importable, so type checkers will not be able to do any validation against it. We may as well just use `torch.Tensor`.